### PR TITLE
Extend JSON support for WAN config classes

### PR DESCRIPTION
**Pull request by mmedenjak**
Covered all WAN replication config classes:
- WanReplicationConfig
- WanPublisherConfig
- WanConsumerConfig
- AliasedDiscoveryConfig
- DiscoveryConfig
- DiscoveryStrategyConfig
- WanSyncConfig

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/management/dto/WanConsumerConfigDTO.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        config.setProperties(fromJsonObject((JsonObject) json.get("properties")));
+    }
+
+    public WanConsumerConfig getConfig() {
+        return config;
+    }
+
+    private Map<String, Comparable> fromJsonObject(JsonObject object) {
```
Comment (mustafaiman) [inline hazelcast/src/main/java/com/hazelcast/internal/management/dto/WanConsumerConfigDTO.java]: maybe you can extract this and `toJsonObject` methods to some common place and re-use. `JsonUtil` may be appropriate.

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/internal/management/dto/WanConsumerConfigDTO.java]: Done. While writing the `fromJson` and `toJson` methods on various classes I was actually thinking of introducing some JSON serialization annotations and a mechanism to read from and write to an annotated class. But it seemed like an overkill for this PR.

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/management/dto/AliasedDiscoveryConfigDTO.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if (!isNullOrEmpty(config.getProperties())) {
+            root.add("properties", toJsonObject(config.getProperties()));
+        }
+
+        return root;
+    }
+
+    @Override
+    public void fromJson(JsonObject json) {
+        config = AliasedDiscoveryConfigUtils.<C>newConfigFor(tag);
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/management/dto/AliasedDiscoveryConfigDTO.java]: Minor: no need for the explicit type.

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/internal/management/dto/AliasedDiscoveryConfigDTO.java]: Yes, tried to remove that but for some reason I get `incompatible types: inference variable C has incompatible upper bounds com.hazelcast.config.AliasedDiscoveryConfig<C>,C`. My generics foo is not strong enough to solve that :)

**Comment by mmedenjak** [comment]
Thanks for the reviews @mustafaiman @blazember !

**Pull request merged**